std/conv.d

```diff
@@ -3142,9 +3142,9 @@ if (isInputRange!Source && isSomeChar!(ElementType!Source) && !is(Source == enum
             while (isDigit(i))
             {
                 sawDigits = true;        /* must have at least 1 digit   */
-                if (msdec < (0x7FFFFFFFFFFFL-10)/10)
+                if (msdec < (long.max-10)/10)
                     msdec = msdec * 10 + (i - '0');
-                else if (msscale < (0xFFFFFFFF-10)/10)
+                else if (msscale < 10_000_000_000_000_000_000UL)
```

Comment:
Stop at the largest power of 10 that a `ulong` can hold.

Reply:
I think it makes sense to add this as a comment in the source for future readers.